oci/image.bzl

```diff
@@ -70,11 +78,12 @@ oci_image_layer = rule(
             doc = "Dictionary of file -> file location in tarball",
             allow_files = True,
         ),
+        "zstd_compression": attr.bool(
+            doc = "Indicates whether zstd compression should be used.",
+            default = False,
+        ),
```

Comment:
Suggestion: Similar theme to making compression generic: it may be better to just use [`attr.string`](https://bazel.build/rules/lib/toplevel/attr#string) with `values` of `["none", "gzip", "zstd"]`.